File: src/prompts/reconstructPrompt.js
/**
 * Prompt para reconstruir o curriculo completo.
 * Combina o curriculo base, a analise (step1), os projetos do GitHub e os erros encontrados.
 * Retorna JSON estruturado pronto para gerar o .docx.
 */
export function buildReconstructPrompt({ resume, job, step1Result, githubEval, mode }) {
  const jobSection = job
    ? 'VAGA ALVO:\n---\n' + job + '\n---\n'
    : ''

  const analysisSection = step1Result
    ? 'DIAGNOSTICO DA ANALISE:\n' + JSON.stringify(step1Result, null, 2) + '\n'
    : ''

  const githubSection = githubEval
    ? 'AVALIACAO DO GITHUB:\n' + JSON.stringify(githubEval, null, 2) + '\n'
    : ''

  const modeNote = {
    geral: 'Reconstrua para maximo impacto geral.',
    ats: 'Priorize palavras-chave ATS e formatacao limpa.',
    rewrite: 'Reescreva completamente as secoes fracas com linguagem de alto impacto.',
  }[mode] || 'Reconstrua para maximo impacto geral.'

  return [
    'Voce e ResumeForge, especialista em reconstrucao de curriculos tech.',
    modeNote,
    '',
    'CURRICULO ORIGINAL:',
    '---',
    resume,
    '---',
    '',
    analysisSection,
    githubSection,
    jobSection,
    'INSTRUCOES:',
    '- Use o curriculo original como base — preserve dados reais (empresas, datas, formacao)',
    '- Incorpore os projetos destaque do GitHub como secao "Projetos" se nao existir ou for fraca',
    '- Corrija os problemas identificados na analise (frases vagas, falta de metricas, etc)',
    '- Adicione habilidades identificadas no GitHub que nao estao no curriculo original',
    '- Reescreva experiencias com verbos de acao e metricas quando possivel',
    '- Mantenha no maximo 2 paginas de conteudo',
    '- Idiomas e certificacoes: preserve o que existe, adicione apenas se identificado no GitHub',
    '',
    'Retorne SOMENTE este JSON (sem texto adicional, sem markdown):',
    '{',
    '  "nome": "<nome completo>",',
    '  "titulo_profissional": "<ex: Desenvolvedor Full Stack | 3+ anos>",',
    '  "contato": {',
    '    "email": "<email>",',
    '    "telefone": "<telefone ou vazio>",',
    '    "linkedin": "<url linkedin ou vazio>",',
    '    "github": "<url github ou vazio>",',
    '    "portfolio": "<url portfolio ou vazio>"',
    '  },',
    '  "resumo_profissional": "<3-4 frases impactantes sobre o candidato>",',
    '  "experiencias": [',
    '    {',
    '      "empresa": "<nome>",',
    '      "cargo": "<cargo>",',
    '      "periodo": "<mes/ano - mes/ano ou Atual>",',
    '      "descricoes": ["<frase de impacto com verbo de acao>", "..."]',
    '    }',
    '  ],',
    '  "projetos": [',
    '    {',
    '      "nome": "<nome>",',
    '      "tecnologias": "<stack>",',
    '      "descricao": "<1-2 frases sobre o que faz e impacto>",',
    '      "url": "<link github ou deploy>"',
    '    }',
    '  ],',
    '  "habilidades": {',
    '    "linguagens": ["<lista>"],',
    '    "frameworks": ["<lista>"],',
    '    "ferramentas": ["<lista>"]',
    '  },',
    '  "formacao": [',
    '    {',
    '      "instituicao": "<nome>",',
    '      "curso": "<nome do curso>",',
    '      "periodo": "<ano inicio - ano fim ou Em andamento>"',
    '    }',
    '  ],',
    '  "idiomas": [',
    '    { "idioma": "<nome>", "nivel": "<Nativo | Fluente | Avancado | Intermediario | Basico>" }',
    '  ],',
    '  "certificacoes": ["<nome da cert (plataforma, ano)>"]',
    '}',
  ].join('\n')
}

File: backend/prompts/reconstruct_prompt.py
import json

def build_reconstruct_prompt(resume: str, job: str, mode: str,
                              step1_result: dict | None, github_eval: dict) -> str:
    problems = step1_result.get("problemas", []) if step1_result else []
    highlights = github_eval.get("projetos_destaque", [])
    skills = github_eval.get("habilidades_identificadas", [])
    strong_points = github_eval.get("pontos_fortes", [])

    job_section = f"\nVAGA ALVO:\n{job}\n" if job and job.strip() else ""

    return f"""Você é um redator especialista em currículos para mercado tech. Reconstrua o currículo abaixo incorporando as informações dos projetos do GitHub.

CURRÍCULO ORIGINAL:
{resume}
{job_section}
PROBLEMAS IDENTIFICADOS NA ANÁLISE:
{json.dumps(problems, ensure_ascii=False)}

PROJETOS DESTACADOS DO GITHUB:
{json.dumps(highlights, ensure_ascii=False, indent=2)}

HABILIDADES IDENTIFICADAS NO GITHUB:
{json.dumps(skills, ensure_ascii=False)}

PONTOS FORTES DO GITHUB:
{json.dumps(strong_points, ensure_ascii=False)}

REGRAS DE RECONSTRUÇÃO:
1. Máximo 2 páginas — seja conciso e de alto impacto.
2. Adicione os projetos relevantes do GitHub na seção de projetos.
3. Corrija os problemas identificados com linguagem específica e métricas quando possível.
4. Mantenha todas as informações verdadeiras do currículo original.
5. Use verbos de ação no passado para experiências (desenvolveu, implementou, liderou).
6. Se houver vaga alvo, otimize keywords para ela.

Retorne SOMENTE o JSON, sem markdown:
{{
  "nome": "<nome completo>",
  "titulo": "<cargo/título profissional>",
  "contato": {{
    "email": "<email>",
    "telefone": "<telefone ou null>",
    "linkedin": "<url ou null>",
    "github": "<url ou null>",
    "localizacao": "<cidade/estado ou null>"
  }},
  "resumo_profissional": "<3-4 frases de alto impacto sobre o profissional>",
  "experiencia": [
    {{
      "cargo": "<cargo>",
      "empresa": "<empresa>",
      "periodo": "<mês/ano — mês/ano ou atual>",
      "responsabilidades": ["<bullet 1>", "<bullet 2>"]
    }}
  ],
  "projetos": [
    {{
      "nome": "<nome do projeto>",
      "descricao": "<1-2 frases descrevendo o projeto e impacto>",
      "tecnologias": ["<tech1>"],
      "url": "<url ou null>"
    }}
  ],
  "habilidades": {{
    "linguagens": ["<lang1>"],
    "frameworks": ["<fw1>"],
    "ferramentas": ["<tool1>"],
    "outros": ["<outro1>"]
  }},
  "formacao": [
    {{
      "curso": "<nome do curso>",
      "instituicao": "<nome>",
      "ano_conclusao": "<ano ou previsto ano>"
    }}
  ],
  "idiomas": [
    {{"idioma": "<idioma>", "nivel": "<nível>"}}
  ],
  "certificacoes": ["<cert1>"]
}}"""
